test/ectest.c

```diff
@@ -1598,7 +1598,53 @@ static int parameter_test(void)
     OPENSSL_free(buf);
     return r;
 }
-#endif
+
+static int check_ec_key_field_public_range(int id)
+{
+    int ret = 0, type = 0;
+    const EC_POINT *pub = NULL;
+    const EC_GROUP *group = NULL;
+    const EC_METHOD *meth = NULL;
+    const BIGNUM *field = NULL;
+    BIGNUM *x = NULL, *y = NULL;
+    EC_KEY *key = NULL;
+
+    if (!(TEST_ptr(x = BN_new())
+          && TEST_ptr(y = BN_new())
+          && TEST_ptr(key = EC_KEY_new_by_curve_name(curves[id].nid))
+          && TEST_ptr(group = EC_KEY_get0_group(key))
+          && TEST_ptr(meth = EC_GROUP_method_of(group))
+          && TEST_ptr(field = EC_GROUP_get0_field(group))
+          && TEST_int_gt(EC_KEY_generate_key(key), 0)
+          && TEST_int_gt(EC_KEY_check_key(key), 0)
+          && TEST_ptr(pub = EC_KEY_get0_public_key(key))
+          && TEST_int_gt(EC_POINT_get_affine_coordinates(group, pub, x, y,
+                                                         NULL), 0)
+          && TEST_int_ne(type = EC_METHOD_get_field_type(meth), NID_undef)))
```

Comment:
This doesn't make much sense to me: I believe it would be cleaner to move `EC_METHOD_get_field_type()` out of this chain, just before line 1630; then have:

```C
if (type == NID_X9_62_characteristic_two_field) {
    /* test for bin curves */
} else if (type == NID_X9_62_prime_field) {
    /* test for prime curves */
} else {
    /* this should never happen */
    TEST_true(0);
    goto err;
}
```